### PR TITLE
[threads] Fix some unified suspend bugs

### PR DESCRIPTION
**Pull request by ludovic-henry**

### CODE REVIEW

File: mono/metadata/sgen-stw.c [hunk 1]
```diff
@@ -305,14 +305,16 @@ mono_sgen_init_stw (void)
...
+sgen_is_thread_in_current_stw (SgenThreadInfo *info, int *reason)
 {
 	/*
 	A thread explicitly asked to be skiped because it holds no managed state.
 	This is used by TP and finalizer threads.
 	FIXME Use an atomic variable for this to avoid everyone taking the GC LOCK.
 	*/
 	if (info->client_info.gc_disabled) {
+		if (reason)
+			*reason = 1;
```
Comment (lambdageek) [inline mono/metadata/sgen-stw.c]: Worth making an enum for these reasons, I wonder?

**Comment by alexrp** [comment]
Regarding 095f23628ad8e03f7de51dd3b22f4196dc59b978: while you're here, could you rename that function to `mono_native_thread_yield`? That's the convention we use for thread functions which don't require a current `MonoThreadInfo` to be set.

**Comment by ludovic-henry** [comment]
@kumpera could you please review? This PR doesn't enable unified suspend by default yet, it just fixes a bunch of bugs and typos.

**Comment by ludovic-henry** [comment]
The Linux and ARM failures are unrelated.

File: mono/metadata/sgen-stw.c (hunk 1 above)
Comment (kumpera) [inline mono/metadata/sgen-stw.c]: Yeah, it should be an enum and logging should stringify it.

File: mono/tests/sgen-new-threads-dont-join-stw-2.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (kumpera) [inline mono/tests/sgen-new-threads-dont-join-stw-2.cs]: ❤️

**Comment by kumpera** [comment]
It all looks fine, there are some cosmetic changes that would be nice to be addressed but can be done afterwards.

**Pull request merged**